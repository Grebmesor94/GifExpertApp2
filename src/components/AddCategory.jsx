import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({onAddCategory}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInput = ({target}) => { 
    setInputValue(target.value)
  }

  const handleSubmit = (e) => {
    e.preventDefault() 
    if(inputValue < 1) return;

    onAddCategory(inputValue.toLowerCase().trim())
    setInputValue('')
  }

  return (
    <form 
      onSubmit={handleSubmit}
      aria-label='form'
    > 
      <input 
        type="text"
        placeholder="category"
        value={inputValue}
        onChange={handleInput}
      />
    </form>
  )
}

AddCategory.propTypes = { 
  onAddCategory: PropTypes.func.isRequired
}