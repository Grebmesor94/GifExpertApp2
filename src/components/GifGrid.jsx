
import PropTypes from 'prop-types';
import { useFetch } from '../hooks/useFetch';
import { GifItem } from './GifItem';

export const GifGrid = ({category}) => {

  const { images, loading } = useFetch(category)

  return (
    <>
    
      <h3>{category}</h3>

      {
        loading && <h3>Cargando...</h3>
      }

      <div className='card-grid'>
        {
          images.map((image) => (<GifItem key={image.id} {...image} />))
        }
      </div>

    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}