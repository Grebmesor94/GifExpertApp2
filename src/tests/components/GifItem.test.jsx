import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components";

describe('test on <GifItem />', () => { 
  const titletest = 'titulo';
  const urltest =  'https://prueba.js/';
  
  test('should implement PropTypes', () => {

    const {container} = render(<GifItem  title={titletest} url={urltest}/>)

    expect(container).toMatchSnapshot();

  });

  test('should show url and title described', () => {
    render(<GifItem url={urltest} title={titletest} />)

    // expect(screen.getByRole('img').src).toBe(urltest);
    // expect(screen.getByRole('img').alt).toBe(titletest);

    const {src, alt} = screen.getByRole('img')

    expect(src).toBe(urltest);
    expect(alt).toBe(titletest);

  });

 })