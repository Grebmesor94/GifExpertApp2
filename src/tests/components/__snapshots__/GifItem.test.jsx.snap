// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifItem /> should implement PropTypes 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://prueba.js/"
    />
  </div>
</div>
`;
